app/lib/properCapitalize.js

```diff
@@ -0,0 +1,62 @@
+function capitalize(str) {
+  return str.charAt(0).toUpperCase() + str.slice(1);
+}
+
+function capitalizeWithSpecialCharacter(str) {
+  return str.charAt(0) + str.charAt(1).toUpperCase() + str.slice(2);
+}
+
+function capitalizeAllCaps(str) {
+  return str.charAt(0) + str.slice(1).toLowerCase();
+}
+
+function capitalizeWithSpecialCharacterCaps(str) {
+  return str.charAt(0) + str.charAt(1).toUpperCase() + str.slice(2).toLowerCase();
+}
+
+function capitalizeNameWithApostrophes(str) {
+  let stringArray = '';
+  // eslint-disable-next-line arrow-body-style
+  str.split(/'/).forEach((name, idx, array) => {
+    if (idx === array.length - 1) {
+      if (idx === 1) {
+        // eslint-disable-next-line no-use-before-define
+        stringArray += titleize(name);
+      } else {
+        stringArray += name.toLowerCase();
+      }
+    } else {
+      // eslint-disable-next-line no-use-before-define
+      stringArray += `${titleize(name)}'`;
+    }
+  });
+  return stringArray;
+}
+
+function titleize(input) {
+  let stringArray = input.replace(/ {2}/g, ' ').split(' ');
```

Comment:
Can this be extracted into a function and named `removeAdditionalSpaces` (or something similar) e.g.

```js
return removeAdditionalSpaces(input)
.split(' ')
.map((input) => {
///
})
.join(' ');
```

There is no need to assign the result of `map` to anything either, it can be returned, as long as `join(' ')` is the last function.